model-optimizer/extensions/middle/StridedSliceReplacer.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from extensions.middle.InsertLayoutPropagationTransposes import InsertLayoutPropagationTranspose
+from extensions.middle.StridedSliceNormalizer import StridedSliceNormalizer
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_node_with_second_input
+from mo.graph.graph import Graph, rename_nodes
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.squeeze import Squeeze
+from mo.ops.unsqueeze import Unsqueeze
+
+
+def replace_strided_slice(node, mask, op):
+    node_name = node.soft_get('name', node.id)
+    axes = np.where(mask == 1)[0]
+    new_node = create_op_node_with_second_input(node.graph, op, int64_array(axes))
+    node.in_port(0).get_connection().set_destination(new_node.in_port(0))
+    node.out_port(0).get_connection().set_source(new_node.out_port(0))
+
+    rename_nodes([(node, node_name + '/ShouldBeDeleted'), (new_node, node_name)])
+
+
+class ReplaceStridedSliceWithSqueezeUnsqueeze(MiddleReplacementPattern):
+    r"""
+    The transformation replaces StridedSlice with a Squeeze/Unsqueeze node if StridedSlice executes like a Squeeze/Unsqueeze
+    and does not slice values. This is necessary if StridedSlice is to be executed in original N(D)HWC layout, because
+    the operation does not have reinterp_shape attribute and MO can not insert NC(D)HW -> N(D)HWC Transpose in
+    extensions/middle/InsertLayoutPropagationTransposes.py.
+    """
+    enabled = True
+    force_clean_up = True
```

Comment:
No need to specify `force_clean_up`. The cleanup is quite expensive so we should avoid calling it where possible. Instead, just remove the StridedSlice operation in the `replace_strided_slice` function.